src/evaluator/functions.rs

```diff
@@ -1089,6 +1089,69 @@ pub fn to_millis<'a>(
     }
 }
 
+pub fn fn_zip<'a>(
+    context: FunctionContext<'a, '_>,
+    args: &[&'a Value<'a>],
+) -> Result<&'a Value<'a>> {
+    if args
+        .iter()
+        .any(|arg| arg.is_null() || matches!(arg, Value::Undefined))
+    {
+        return Ok(context.arena.alloc(Value::Array(
```

Comment:
I think we could use the `Value::array(context.arena, ArrayFlags::empty())` factory function.